src/GDBTargetDebugSession.ts

```diff
@@ -535,9 +535,13 @@ export class GDBTargetDebugSession extends GDBDebugSession {
                 this.serialPort.close();
 
             if (this.targetType === 'remote') {
-                if (this.gdb.getAsyncMode() && this.isRunning)
-                    this.gdb.sendCommand('interrupt');
-                await this.gdb.sendCommand('disconnect');
+                if (this.gdb.getAsyncMode() && this.isRunning) {
+                    this.gdb
```

Comment:
I guarantee that this does not fix the problem and your users will still get `Cannot execute this command while the target is running` errors with this change. I will add a disclaimer into the code to explain and look forward to a future update that actually waits for the target to be interrupted before sending the disconnect. I have raised #295 to add some suitable API to make it easier to write this code, perhaps as Microchip is being affected by this you or @parisa-mchp can fix this.

```suggestion
                    // See #295 - this use of "then" is to try to slightly delay the
                    // call to disconnect. A proper solution that waits for the
                    // interrupt to be successful is needed to avoid future
                    // "Cannot execute this command while the target is running"
                    // errors 
                    this.gdb
```

Reply:
Ok. So there is a delay between sending the interrupt signal and the target being actually interrupted. Disconnecting the target will only work if the target is actually interrupted.

Are you worried that in the future somehow the delay gets larger than the interval between sending these two signals, the issue would occur again? If that's the case, we will definitely work on adding the API to support this functionality in the future.